pkg/volume/util/util.go

```diff
@@ -96,29 +97,43 @@ func UnmountPath(mountPath string, mounter mount.Interface) error {
 // IsNotMountPoint will be called instead of IsLikelyNotMountPoint.
 // IsNotMountPoint is more expensive but properly handles bind mounts.
 func UnmountMountPoint(mountPath string, mounter mount.Interface, extensiveMountPointCheck bool) error {
-	if pathExists, pathErr := PathExists(mountPath); pathErr != nil {
+	pathExists, pathErr := PathExists(mountPath)
+	return unmountMountPath(mountPath, mounter, extensiveMountPointCheck, pathExists, pathErr)
+}
+
+// UnmountMountPoint is a common unmount routine that unmounts the given path and
+// deletes the remaining directory if successful.
+// if extensiveMountPointCheck is true
+// IsNotMountPoint will be called instead of IsLikelyNotMountPoint.
+// IsNotMountPoint is more expensive but properly handles bind mounts.
+// pathExists should be true if the mountPath existed,else false
+// pathErr is the error when stats the mountPath
+func unmountMountPath(mountPath string, mounter mount.Interface, extensiveMountPointCheck bool, pathExists bool, pathErr error) error {
+	corruptedMnt := isCorruptedMnt(pathErr)
+	if pathErr != nil && !corruptedMnt {
```

Comment:
Why is this check before the check if pathExists? If the path does not exist what is the reason to check of the mount is corrupted?